07-forms/01-CounterButtonComplete/components/CounterButton.vue

```diff
@@ -1,9 +1,39 @@
 <template>
-  <button></button>
+  <button @click="handleClick">{{ state }}</button>
 </template>
 
 <script>
 export default {
   name: 'CounterButton',
+
+  props: {
+    count: {
+      type: Number,
+      default: 0,
+    },
+  },
+
+  emits: ['update:count'],
+
+  data() {
+    return {
+      state: this.count,
+    };
+  },
+
+  watch: {
+    count(newValue) {
+      if (this.state !== newValue) {
+        this.state = newValue;
+      }
```

Comment:
Checking that values are different is redundant here. 

It was important with objects, because of cloning: he had new property value, new object reference.